components/CheckDocumentation/vacuum/XminHorizon.tsx

```diff
@@ -0,0 +1,342 @@
+        connection's pid by running the following command:
+      </p>
+      <CodeBlock>
+        <SQL
+          sql={`SELECT pid, datname, usename, state, backend_xmin, backend_xid
+                  FROM pg_stat_activity
+                  WHERE backend_xmin IS NOT NULL OR backend_xid IS NOT NULL
+                  ORDER BY greatest(age(backend_xmin), age(backend_xid)) DESC;`}
+        />
+      </CodeBlock>
+      <p>You can cancel it by running either of commands:</p>
+      <CodeBlock>
+        <SQL
+          sql={`SELECT pg_cancel_backend('<query_pid>');
+                SELECT pg_terminate_backend('<query_pid>');`}
+        />
+      </CodeBlock>
+    </li>
+  );
+};
+
+const GuidanceByReplicationSlot: React.FunctionComponent<{
+  inApp: boolean;
+  heldBackInfo: number | null;
+  serverReplicationUrl: string;
+}> = ({ inApp, heldBackInfo, serverReplicationUrl }) => {
+  if (inApp && !heldBackInfo) {
+    return null;
+  }
+
+  const Link = useSmartAnchor();
+  const CodeBlock = useCodeBlock();
+  return (
+    <li>
+      <h5>Lagging or stale physical replication slots</h5>
+      <p>
+        With physical streaming replication with{" "}
+        <code>hot_standby_feedback</code> is on, when replication is lagging or
+        a replica server is stale (e.g. down, gone), the oldest transaction that
+        the replication slot needs the database to retain can be "stuck",
+        holding back the xmin horizon.
+      </p>
+      <h6>
+        <b>Solution</b>
+      </h6>
+      {heldBackInfo && (
+        <p>
+          A replication slot is holding back the xmin horizon at{" "}
+          <code>{heldBackInfo["xmin"]}</code>.
+        </p>
+      )}
+      <p>
+        You can check the replication status on the{" "}
+        <Link to={serverReplicationUrl}>Replication</Link> page.
+      </p>
+      <p>
+        If the replication slot is no longer used, you can remove it by running
+        the following command:
+      </p>
+      <CodeBlock>
+        <SQL sql={`SELECT pg_drop_replication_slot('<slot_name>');`} />
+      </CodeBlock>
+    </li>
+  );
+};
+
+const GuidanceByReplicationSlotCatalog: React.FunctionComponent<{
+  inApp: boolean;
+  heldBackInfo: number | null;
+  serverReplicationUrl: string;
+}> = ({ inApp, heldBackInfo, serverReplicationUrl }) => {
+  if (inApp && !heldBackInfo) {
+    return null;
+  }
+
+  const CodeBlock = useCodeBlock();
+  const Link = useSmartAnchor();
+  return (
+    <li>
+      <h5>Lagging or stale logical replication slots</h5>
+      <p>
+        With logical replication slots, replication can also get stale when DDL
+        changes (database migrations) don't get applied to a replica server
+        (subscriber). When the subscriber was unable to replicate data due to a
+        schema mismatch, replication will error and get stale. This causes the
+        system catalogs xmin of the primary (publisher) to be held back until
```

Comment:
```suggestion
        xmin of the system catalogs of the primary (publisher) to be held back until
```
I think the triple compound noun here is kind of awkward. I admit the rewording is not great either, but I think it's a small step up. If you can think of other ways to reword that, that can work, too.